gammapy/cube/make.py

```diff
@@ -257,3 +260,125 @@ def _check_selection(selection):
             raise ValueError("Selection not available: {!r}".format(name))
 
     return selection
+
+class IrfMapMaker:
+    """Make IRF maps from IACT observations.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.WcsGeom`
+        Reference map geometry with true energy axis
+    offset_max : `~astropy.coordinates.Angle`
+        Maximum offset angle
+    rad_axis : `~gammapy.maps.MapAxis`
+        The rad axis of the PsfMap to be built
+    migra_axis: `~gammapy.maps.MapAxis`
+        The migra axis of the EDispMap to be built
+    """
+
+    def __init__(self, geom, offset_max, rad_axis=None, migra_axis=None):
+        if not isinstance(geom, WcsGeom):
+            raise ValueError("IrfMapMaker only works with WcsGeom")
+
+        if geom.is_image:
+            raise ValueError("IrfMapMaker only works with geom with an energy axis")
+
+        self.geom = geom
+        self.offset_max = Angle(offset_max)
+
+        if rad_axis is None and migra_axis is None:
+            raise ValueError("No rad nor migra axis passed to IrfMapMaker")
+
+        self.rad_axis = rad_axis
+        self.migra_axis = migra_axis
+
+        self.maps = {}
```

Comment:
In the `MapMaker` we have found this to be problematic pattern, because for joint analysis the `MapMaker` cannot be reused but must be created every-time again in the loop over observations. There are multiple solutions: either reset `self.maps` in the `.run()` method to an empty dict, or remove `self.maps` and just create a local variable in `.run()` that is returned (which I'd prefer...)